pkg/urbit/jets/c/repn.c

```diff
@@ -0,0 +1,112 @@
+#include "all.h"
+
+/*
+  Get the lowest `n` bits of a word `w` using a bitmask.
+*/
+#define TAKEBITS(n,w) \
+  ((n)==32) ? (w) :   \
+  ((n)==0)  ? 0   :   \
+  ((w) & ((1 << (n)) - 1))
+
+/*
+  Divide, rounding up.
+*/
+#define DIVCEIL(x,y) \
+  (x==0) ? 0 :       \
+  1 + ((x - 1) / y);
+
+u3_noun u3qc_repn(u3_atom bits, u3_noun blox) {
+    if ( !_(u3a_is_cat(bits) || bits==0 || bits>31) ) {
+        return u3m_bail(c3__fail);
+    }
+
+    //
+    //  Calculate input and output size.
+    //
+    c3_w num_blox = u3qb_lent(blox);
+    c3_w bit_widt = num_blox * bits;
+    c3_w wor_widt = DIVCEIL(bit_widt, 32);
```

Comment:
I won't hold you to CVC, but I must insist on the "hungarian" suffixes for all non `u3_noun`/`u3_atom` variables.